examples/bsoncxx/view_and_value.cpp

```diff
@@ -0,0 +1,113 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <vector>
+
+#include <bsoncxx/array/view.hpp>
+#include <bsoncxx/array/view.hpp>
+#include <bsoncxx/builder/basic/array.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/stdx/string_view.hpp>
+#include <bsoncxx/types.hpp>
+
+using namespace bsoncxx;
+
+int main(int, char**) {
+    // This example will cover the read-only BSON interface.
+
+    // Lets first build up a non-trivial BSON document using the builder interface.
+    using builder::basic::kvp;
+    using builder::basic::sub_array;
+
+    auto doc = builder::basic::document{};
+    doc.append(kvp("team", "platforms"),
+               kvp("id", types::b_oid{oid(oid::init_tag)}),
+               kvp("members", [](sub_array sa) {
+                   sa.append("tyler", "jason", "drew",
+                             "sam", "ernie", "john",
+                             "mark", "crystal");
+               })
+    );
+
+    // document::value is an owning bson document conceptually similar to string.
+    document::value value{doc.extract()};
+    // document::view is a non-owning bson document conceptually similar to string_view.
+    document::view view{value.view()};
+    // array::view and array::value are the corresponding classes for arrays.
+
+    // we can print a view using to_json
+    std::cout << to_json(view) << std::endl;
+
+    // note that all of the interesting methods for reading BSON are defined on the view type.
+
+    // iterate over the elements in a bson document
+    for (document::element ele : view) {
```

Comment:
Can you say auto here, or did you want to say element explicitly to demonstrate the type?